starknet_py/cairo/serialization/data_serializers/named_tuple_serializer.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import Union, Dict, NamedTuple, OrderedDict, Generator
+
+from starknet_py.cairo.serialization._context import (
+    DeserializationContext,
+    SerializationContext,
+)
+from starknet_py.cairo.serialization.data_serializers._common import (
+    deserialize_to_dict,
+    serialize_from_dict,
+)
+from starknet_py.cairo.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.utils.tuple_dataclass import TupleDataclass
```

Comment:
What's the purpose of `TupleDataclass`?

Reply:
It is a better replacement for the Result. We pretended it was a NamedTuple (it wasn't :|)

Reply:
Oh, it's that thing. Makes sense.

Reply:
The name is okay I think.